### PR TITLE
Add support for fetching the site url

### PR DESCRIPTION
**Pull request by garemoko**
See https://github.com/jlowe64/moodle-logstore_xapi/issues/25

Please wait until I get PRs up for all three modules before merging.

### CODE REVIEW

**Comment by garemoko** [comment]
Also i need to do tests and documentation still

**Comment by garemoko** [comment]
This is now failing, which is actually an improvement :-).

**Comment by garemoko** [comment]
@jlowe64 @ryansmith94 this is ready for review. Thanks!

**Comment by ryansmith94** [comment]
Again thanks for updating the docs and tests, good stuff in 6aac0d8. I think I made a comment about that on the issue we were discussing. All seems good, my only concern is that I think the `assertSite` needs more assertions.

**Comment by garemoko** [comment]
This next commit should cause the tests to fail if I get it right.

**Comment by garemoko** [comment]
@ryansmith94 what does `$this->assertEquals($input, $output->id);` do? It's failing on that.

**Comment by garemoko** [comment]
I'm removing `$this->assertEquals($input, $output->id);` because I don't think it's relevant for this object type. The site id doesn't need to match the input (it doesn't have an id) and we don't make that check for assert course.

File: tests/EventTest.php [hunk 1]
```diff
@@ -71,6 +71,11 @@ private function assertCourse($input, $output) {
         $this->assertEquals($this->cfg->wwwroot . '/course/view.php?id=' . $output->id, $output->url);
     }
 
+    private function assertSite($input, $output) {
+        $this->assertEquals($this->cfg->wwwroot, $output->url);
+        $this->assertEquals('site', $output->type);
```
Comment (ryansmith94) [inline tests/EventTest.php]: `type` in the examples is set to `mod_course` not `site`.

**Comment by ryansmith94** [comment]
That's fine @garemoko, it's because it was previously referring to the first course which had an `id` of `1`.

**Comment by garemoko** [comment]
Docs updated as requested. I think some of the types for other elements are also wrong, but that's out of scope of this PR. IMHO it's worth reducing the number of examples for reasons of maintainability. It's been a lot of work just to add two context activities.

**Comment by ryansmith94** [comment]
Super thanks @garemoko. Possibly, but yeah out of scope here like you say. I agree that we need to improve the maintainability, however I think the examples are super valuable to users (particularly new ones) and also devs (I find myself checking them too). I think what we actually need is someway to generate them.

**Pull request merged**